registry/client/repository.go

```diff
@@ -0,0 +1,657 @@
+package client
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"time"
+
+	ctxu "github.com/docker/distribution/context"
+
+	"github.com/docker/distribution/manifest"
+
+	"github.com/docker/distribution/digest"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/registry/api/v2"
+	"golang.org/x/net/context"
+)
+
+// NewRepositoryClient creates a new Repository for the given repository name and endpoint
+func NewRepositoryClient(ctx context.Context, name string, endpoint *RepositoryEndpoint) (distribution.Repository, error) {
```

Comment:
@dmcgowan I guess this could be called NewRepository since it's returning a new repository. wdyt?